JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/slurm/job/JobSubmission.java

```diff
@@ -439,6 +443,17 @@ public void monitorJobs() throws SlurmJobException{
 				session = jsch.getSession(slurmJobProperty.getHpcServerLoginUserName(), getHpcAddress(), 22);
 				String pwd = slurmJobProperty.getHpcServerLoginUserPassword();
 				session.setPassword(pwd);
+
+				try {
+					Connector con = new PageantConnector();
+					IdentityRepository irepo = new RemoteIdentityRepository(con);
+					jsch.setIdentityRepository(irepo);
+					session.setConfig("PreferredAuthentications", "publickey,keyboard-interactive,password");
+				} catch (AgentProxyException e) {
+					session.setConfig("PreferredAuthentications", "password");
```

Comment:
Same here, wrt hardcoding. You can see hardcoded string repetition starting to occur pretty quickly.

Reply:
Similar to the previous comment, if connecting to Pagaent has failed then the only option left is to authenticate using the password as trying any other method would just fail. Therefore giving the user the option to modify this value seems pointless. I will add some comments to make this clearer in the code.

Reply:
Again, I haven't mentioned anything about user previously.